connection.js

```diff
@@ -855,158 +830,131 @@ class Connection {
+            let timeNumber = 0;                // for example 600 (600 seconds)
+            let timeToString = "";             // for example "10 minutes"  = timespanToDisplayString(config.xxx)
+            let timeNotificationToString = ""; // for example user wanted "1 seconds" = timespanToDisplayString(t.xxx)
+
+            // note: if t.speed === blitzLiveCorr test below is not true we 
+            // exit the function : we dont reject the challenge
+
+            // since the processing after the tests is the same 
+            // for min and max args (only difference is ">" or "<" in the tests)
             // => we store all the processing in sub functions as well :
             function UHMAEATForMaintimesFischerRejectProcessing() {
-                timecontrolSentence = "Initial Time and/or Max Time ";
+                timecontrolSentence = "Initial Time and/or Max Time";
                 timeNotificationToString = timespanToDisplayString(t.initial_time);
-                endingSentence = ".";
-                // example : connSentence + "main time 5 seconds, it is below minimum main time live in byoyomi : 1 minutes"
                 return UHMAEATForMaintimesGenericReject();
             }
-            function UHMAEATForMaintimesByoyomiRejectProcessing() {
-                timecontrolSentence = "Main Time ";
+            // for maintimes, japanese byoyomi and canadian byoyomi use the exact
+            // same formula, so using the same function for both timecontrols
+            function UHMAEATForMaintimesByoyomiOrCanadianRejectProcessing() {
+                timecontrolSentence = "Main Time";
                 timeNotificationToString = timespanToDisplayString(t.main_time);
-                endingSentence = ".";
-                return UHMAEATForMaintimesGenericReject();
-            }
-            function UHMAEATForMaintimesCanadianRejectProcessing() {
-                timecontrolSentence = "Main Time ";
-                timeNotificationToString = timespanToDisplayString(t.main_time);
-                endingSentence = ".";
                 return UHMAEATForMaintimesGenericReject();
             }
             function UHMAEATForMaintimesAbsoluteRejectProcessing() {
-                timecontrolSentence = "Total Time ";
+                timecontrolSentence = "Total Time";
                 timeNotificationToString = timespanToDisplayString(t.total_time);
-                endingSentence = ".";
                 return UHMAEATForMaintimesGenericReject();
             }
+
+            // FINAL UNIVERSAL REJECT :
+            function UHMAEATForMaintimesGenericReject() {
+                conn_log(`${user.username} wanted ${argIsMinMax.minimumMaximum} ${timecontrolSentence} ${t.speed} ${rankedUnranked} ${timeNotificationToString} in ${t.time_control}, it is below ${timeToString}`);
+                // example : "userX wanted minimum maintime blitz ranked 5 seconds,
+                // it is below 1 minutes"
```

Comment:
The example doesn't seem to match the code, ` in ${t.time_control}` is nowhere to be found in the examlpe.

Reply:
indeed ! 
fixed :+1: